backend/app/controllers/spree/admin/users_controller.rb

```diff
@@ -153,6 +155,19 @@ def set_stock_locations
             Spree::StockLocation.accessible_by(current_ability).where(id: user_params[:stock_location_ids])
         end
       end
+
+      def address_valid?
+        bill_address_params = user_params[:bill_address_attributes]
+        ship_address_params = user_params[:ship_address_attributes]
+
+        if(Address.new(bill_address_params).valid? &&
+          Address.new(ship_address_params).valid?)
```

Comment:
I don't _think_ that this approach makes sense given that the attributes are immutably merged with the existing address. I'm pretty sure that's how that works, anyway.

Reply:
With this approach we do not have error messages on the addresses that we could display.

I am puzzling with

```rb
@user.update(user_params) && @user.bill_address.valid? && @user.ship_address.valid?
```

but it does not seem to work either.

Reply:
We use `nested_attributes` for users ship_address and bill_address, but since address a immutable it silently (thanks rails) rejects. The address is always valid, because it never updates. We need a way to use `Spree::Address.immutable_merge` on an existing `user.x_address` instances somehow.

Reply:
I also believe we should add a `Admin::Users::AddressesController` to directly work on the address records in stead of proxying through the user.

Reply:
> Admin::Users::AddressesController

Yes, I think better update address by callling new AddressesController.